test/EndToEndTests/Tests/Client/AsynchronousTests/AsynchronousQueryTests.cs

```diff
@@ -294,21 +295,25 @@ public void ExecuteBatchWithSaveChangesOptionsReturnsCorrectResults()
             this.EnqueueTestComplete();
         }
 
-#if !PORTABLELIB && !NETCOREAPP1_0
         /// <summary>
         /// CancelRequest for Batch Requests
         /// </summary>
-        // github issuse: #896
-        // [Fact, Asynchronous]
+        [Fact, Asynchronous]
         public void CancelBatchRequestTest()
         {
             var context = this.CreateWrappedContext<DefaultContainer>().Context;
             ServicePoint servicePoint = null;
             context.SendingRequest2 += ((sender, args) =>
             {
-                if (args.RequestMessage is HttpClientRequestMessage)
+                servicePoint = ServicePointManager.FindServicePoint(args.RequestMessage.Url);
+                MethodInfo ReleaseConns = servicePoint.GetType().GetMethod
+                        ("ReleaseAllConnectionGroups",
+                        BindingFlags.Instance | BindingFlags.NonPublic);
+                ReleaseConns.Invoke(servicePoint, null);

```

Comment:
Just checking whether I understand what's going on here: You're destroying all connections that currently exist to ensure that you have a clean state before making the batch request so that you can ensure that the request is the only connection when you're performing the assert? Is this correct?

Reply:
this is true